nntrainer/cl_context.cpp

```diff
@@ -86,20 +84,13 @@ static void registerer(ClContext &cc) noexcept {
 };
 
 ClContext &ClContext::Global() {
-  static ClContext instance;
-
   // initializing commandqueue and context
-  bool result = instance.clInit();
-
-  if (!result) {
+  if (!clInit()) {
     ml_loge("cl_context: opencl command queue creation failed");
   }
 
-  /// in g++ there is a bug that hangs up if caller throws,
-  /// so registerer is noexcept although it'd better not
-  /// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70298
-  std::call_once(global_cl_context_init_flag, registerer, std::ref(instance));
-  return instance;
+  registerer(*this);
+  return *this;
```

Comment:
When I proposed the engine and context, I haven't have the time to consider the cl_context. Thanks for taking care of this.